### PR TITLE
Allow create user through sign in

### PR DESCRIPTION
**Pull request by Philinphiladelphia**
## Issue Number

N/a

## Purpose/Implementation Notes

This change allows the user to create their account through the sign-in button. If the user doesn't already exist, the proper calls will be made to create them.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: client/src/pages/account/index.js [hunk 1]
```diff
@@ -1,16 +1,65 @@
...
+    decodeURI(`http://${req.headers.host}${req.url}`)
   )
+
+  const initialProps = {}
+
+  initialProps.orcid = response.response.orcid
+  initialProps.accessToken = response.response.access_token
+  initialProps.refreshToken = response.response.refresh_token
+
+  return initialProps
```
Comment (davidsmejia) [inline client/src/pages/account/index.js:62]:
```suggestion
  const orcidRequest = await api.user.getORCID(
    query.code,
    decodeURI(`http://${req.headers.host}${req.url}`)
  )

  if (orcidRequest.isOk) {
    const {
      orcid,
      access_token: accessToken,
      refresh_token: refreshToken
    } = orcidRequest.response

    return {
      orcid,
      accessToken,
      refreshToken
    }
  }
```

File: client/src/api/index.js [hunk 2]
```diff
@@ -147,6 +148,7 @@ export default {
       return [tokenRequest, userRequest]
     },
     create: async (orcid, accessToken, refreshToken, email, grants) => {
+      console.log(orcid, accessToken, refreshToken, email, grants)
```
Comment (davidsmejia) [inline client/src/api/index.js:151]: I think we should start removing console.logs at this point - I'm going to be turning on the lint rule soon

**Review by davidsmejia — APPROVED** [review]
some small changes but looks good!

**Review by kurtwheeler — APPROVED** [review]
Backend stuff looks good to me :+1:

**Pull request merged**